src/Services/Catalog/Catalog.API/01-Services/ICatalogService.cs

```diff
@@ -1,54 +1,93 @@
+    CatalogItemDataResult GetAllItems(CatalogItemFilter filter);
+}
+
+public class CatalogService(ApplicationDataContext context) : ICatalogService
+{
+    private readonly ApplicationDataContext _context = context;
 
-    public class CatalogService(ApplicationDataContext context) : ICatalogService
+    public IEnumerable<CatalogCategoryDTO> GetAllCategories()
     {
-        private readonly ApplicationDataContext _context = context;
-
-        public CatalogItemDataResult GetAll(CatalogItemFilter filter)
-        {
-            filter ??= new CatalogItemFilter() { PageIndex = 0, PageSize = 10, ShowOnlyHighlighted = true };
-
-            var totalItems = _context.CatalogItems.Count(w => w.IsHighlighted == filter.ShowOnlyHighlighted);
-
-            var data = _context.CatalogItems
-                .Where(w => w.IsHighlighted == filter.ShowOnlyHighlighted)
-                .Skip(filter.PageIndex * filter.PageSize)
-                .Take(filter.PageSize)
-                .AsNoTracking()
-                .Select(item => new CatalogItemDTO
-                {
-                    Id = item.Id,
-                    Name = item.Name,
-                    Description = item.Description,
-                    Brand = item.Brand.Name,
-                    Price = item.Price
-                });
-
-            return new CatalogItemDataResult() { TotalItems = totalItems, Items = data };
-        }
+        return _context.CatalogCategories.Select(c => new CatalogCategoryDTO { Id = c.Id, Name = c.Name });
     }
 
-    public class CatalogItemFilter
+    public CatalogCategoryDTO GetCategoryById(int id)
     {
-        public bool ShowOnlyHighlighted { get; set; }
-        public int PageIndex { get; set; }
-        public int PageSize { get; set; }
+        var category = _context.CatalogCategories.FirstOrDefault(c => c.Id == id);
+
+        if (category is null)
+            return null;
+
+        return new CatalogCategoryDTO { Id = category.Id, Name = category.Name };
     }
 
-    public class CatalogItemDataResult
+    public CatalogItemDataResult GetAllItems(CatalogItemFilter filter)
     {
-        public int TotalItems { get; set; }
-        public IEnumerable<CatalogItemDTO> Items { get; set; }
+        filter ??= new CatalogItemFilter();
+
+        if (filter.PageSize <= 0)
+            filter.PageSize = 10;
+
+        if (filter.PageSize > 50)
+            filter.PageSize = 50;
+
+        var query = _context.CatalogItems.AsQueryable();
+
+        if (filter.ShowOnlyHighlighted) // Aqui o comportamento foi mudado,
+                                        // antes só mostrava produtos com destaque ou sem, eles não mostravam-se juntos.
+            query = query.Where(w => w.IsHighlighted);
+
+        var category = GetCategoryById(filter.CategoryId);
+
+        if (category is not null)
+            query = query.Where(w => w.Category.Id == category.Id);
+
+        var totalItems = query.Count();
```

Comment:
Posso estar errado, mas aqui vale uma verificação adicional. Ao executar o query.Count() tenho a impressão de que ele irá executar a query no banco. Se isso for verdade, o Skip / Take não acontece mais no banco de dados e sim em memoria. Se isso realmente acontece é um problema grande de performance. Precisa validar os eventos gerados do EF durante o Debug para ter certeza que ele só executa um Count(*) e não traz o catalogo inteiro (o where é levado em consideracao) em memoria.

Caso não tenha como validar, sugiro manter como estava, pelo menos assim vc tem a certeza absoluta que ele vai uma vez pro count e outra pros registros ... e nao que ele vai uma vez só e filtra em memoria.